services/app-graphql/src/schema.graphql

```diff
@@ -1396,6 +1435,23 @@ type ContractPackageSubmission {
     rateRevisions: [RateRevision!]!
 }
 
+"""
+ContractPackageSubmission is a snapshot of a contract and all its related rates in time
+a ContractPackageSubmission can be created by the user submitting a contract or a related rate
+"""
+type UnlockedContractPackageSubmission {
```

Comment:
Open question, fields that aren't required in the domain modal (cause for packagesubmission and state for contract) were causing a `cannot update non nullable` error when `unlockContract` called on the frontend. I made those fields optional in the graphql schema but I'm not sure if that's the correct way to address this. Any ideas welcome

Reply:
can you give more details on the error? We're not sending this data from app-web so I only expect it to really crop up on returning the model in the resolvers

Reply:
It's an apollo error `Cannot return null for non-nullable field ContractPackageSubmission.cause.`

<img width="1447" alt="Screenshot 2024-07-01 at 10 36 42 AM" src="https://github.com/Enterprise-CMCS/managed-care-review/assets/67110378/12986091-2b64-49e1-8bd2-ee2ce72be54a">

Reply:
Ah, OK I think I have a guess at the problem. We have a custom resolver for packageSubmissions that adds `cause` to the response. It's not something that's in the domain model b/c it's computed. It looks like since you made a new type here for the UnlockedContractPackageSubmission we're not invoking that resolver so we're skipping adding that field. The most straightforward fix I see would be to reuse the other ContractPackageSubmission type, triggering the resolver we wrote for that, but I'm curious if this is the reason you broke it out as its own type and I'm missing something.

Reply:
Ah sorry for the confusion. This is the error I see when I use `ContractPackageSubmission`. That's why I created `UnlockedContractPackageSubmission` because it works when the cause field is allowed to be null

Reply:
@macrael I got this working, I realized I needed to make a `UnlockedContract` resolver for the resolver chain. It does what the `Contract` resolver does in adding the cause and state fields

Reply:
ah perfect! yes I was thinking about that as I fell asleep last night but forgot this morning, glad you sorted it out.